tempodb/blocklist/poller.go

```diff
@@ -239,19 +254,114 @@ func (p *Poller) pollTenantBlocks(ctx context.Context, tenantID string) ([]*back
 
 	for _, blockID := range blockIDs {
 		bg.Add(1)
-		go func(uuid uuid.UUID) {
+		go func(id uuid.UUID) {
+			defer bg.Done()
+
+			if p.cfg.PollJitterMs > 0 {
+				time.Sleep(time.Duration(rand.Intn(p.cfg.PollJitterMs)) * time.Millisecond)
+			}
+
+			m, cm, pollBlockErr := p.pollBlock(ctx, tenantID, id)
+			if m != nil {
+				chMeta <- m
+			} else if cm != nil {
+				chCompactedMeta <- cm
+			} else if pollBlockErr != nil {
+				anyError.Store(err)
+			}
+		}(blockID)
+	}
+
+	bg.Wait()
+	close(chMeta)
+	close(chCompactedMeta)
+
+	if err = anyError.Load(); err != nil {
+		metricTenantIndexErrors.WithLabelValues(tenantID).Inc()
+		return nil, nil, err
+	}
+
+	return p.flushMetaChannels(chMeta, chCompactedMeta)
+}
+
+func (p *Poller) quickPollTenantBlocks(
+	ctx context.Context,
+	tenantID string,
+	previous backend.Blocklist,
```

Comment:
It felt like a clean way not to modify the old code and be able to fall back to existing behavior.  We can merge them into one if we want, I just didn't want to modify the existing in case we want to make it optional or some other approach.

Reply:
same thought here. we should have just one way of doing this (the new way) to reduce code complexity

Reply:
I've refactored this out so that we only have one code path now.  Need to stare at some tests for a while until they pass again.